### PR TITLE
Init docker for saw-remote-api

### PR DESCRIPTION
**Pull request by LisannaAtGalois**
Inits dockerfile for saw-remote-api. Tested that `"method":"SAW/Cryptol/load module","params":{"module name":"Cryptol","state":null}` works OK.

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -13,8 +13,6 @@ jobs:
           git -C deps/abcBridge submodule update --init
       - run: .github/ci.sh set_version
         id: outputs
-      - uses: satackey/action-docker-layer-caching@v0.0.8
```
Comment (jared-w) [inline .github/workflows/docker.yml:16]: I've had to remove this in other actions, so I went ahead and removed it in this one. Eventually we probably will want to look around and see if there's a better way to do docker + pushing to github somehow and really diving into that. But for now, this'll work.

**Review by jared-w — APPROVED** [review]

File: .github/workflows/docker.yml (hunk 1 above)
Comment (LisannaAtGalois) [reply .github/workflows/docker.yml:16]: I've been thinking that caching docker builds would be a great use case for GitHub container registry - especially given that:

> All data transferred out, when triggered by GitHub Actions, and data transferred in from any source is free.

Plus they are probably going to be colocated way better than e.g. caching to dockerhub.

Comment (jared-w) [reply .github/workflows/docker.yml:16]: That's what I was thinking :)

It'll be a good rabbit hole for me to look into next time I nerd snipe myself

**Review by atomb — APPROVED** [review]
I think this looks good. From what I understand, it'll behave exactly like the other Docker image in terms of when it's built and uploaded. Have I understood that correctly?

**Pull request merged**